internal/services/containers/kubernetes_fleet_manager_resource_gen.go

```diff
@@ -51,6 +49,7 @@ func (r KubernetesFleetManagerResource) Arguments() map[string]*pluginsdk.Schema
 		},
 		"resource_group_name": commonschema.ResourceGroupName(),
 		"hub_profile": {
+			Deprecated: "`hub_profile` is currently not functional and is not be passed to the API. This preview feature isn't meant for production use. More information can be found in the https://learn.microsoft.com/en-us/azure/kubernetes-fleet/architectural-overview",
```

Comment:
This sentence contradicts itself? is it not functional/does nothing (in preview)  or is it a preview feature that should be used?

also i don't see hub profile mentioned there? i'm looking more for documentation (or a gh issue) hub profile is ignored by the API/does nothing and has been deprecated

Reply:
1. hub_profile is a required field in the **Fleet preview API version used by Terraform today**. Customers can only create a fleet with a hub, and can't create a fleet without a hub via this API.

2. hub_profile doesn't exist in the **most recent Fleet GA API version** (as hub is a preview feature). Customers can only create a fleet without a hub, and can't create a fleet with a hub via this API.

3. hub_profile is an optional field in the **most recent Fleet preview API version**. Customers can create a fleet with a hub, or a fleet without a hub via this API.

The PR is trying to change from 1 to 2 as we learned that Terraform should only support GA API versions.